lib/puppet/pops/evaluator/runtime3_resource_support.rb

```diff
@@ -10,6 +10,12 @@ def self.create_resources(file, line, scope, virtual, exported, type_name, resou
     env = scope.environment
     #    loader = Adapters::LoaderAdapter.loader_for_model_object(o, scope)
 
+    if env.configured_path && env.resolved_path && env.configured_path != env.resolved_path
+      if file.start_with?(env.resolved_path)
+        file = File.join(env.configured_path, file.delete_prefix(env.resolved_path))
```

Comment:
Things make sense to me up until this point... I dont understand why we need to Join the second argument here. Can you give me an example of what the strings look like at this point? 

I guess what i'm wondering is why is this not as simple as just something like:
```ruby
if report_configured_environmentpath
  env.configured_path
else
  env.resolved_path
end
```

Reply:
What we want to is update the `file` here which will point to a Puppet manifest, ie:
`/opt/puppetlabs/server/data/puppetserver/filesync/client/versioned-dirs/puppet-code/production-1234567890asdf/apache/manifests/vhosts.pp`

So `env.resolved_path` will be something like `/opt/puppetlabs/server/data/puppetserver/filesync/client/versioned-dirs/puppet-code/production-1234567890asdf`
and `env.configured_path` will be something like `/etc/puppetlabs/puppetserver/code/production`.

We want to report `/etc/puppetlabs/puppetserver/code/production/apache/manifests/vhosts.pp` since that file path won't change with every new deploy of production.

Reply:
ah, thanks that makes sense. Not quite sure I fully understand the cases where an `env` could not have a `{configured, resolved}_path` but being defensive makes sense. Thanks for providing the examples that really helped see what the intent is here.

Reply:
Yeah, we should always being going through the Environment loaders in Puppet Server, but I need to double check that's the case when running `puppet apply` and when loaded as a library in puppet-rspec, etc. Figured being defensive was worth it here.